### PR TITLE
Problem: contract state export to genesis is not tested

### PR DESCRIPTION
**Pull request by yihuang**
Closes: #103

- export contract state in genesis and start devnet with it.
- check contract state
- fix the genesis export issue in ethermint

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: integration_tests/test_exported_genesis.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+@pytest.fixture(scope="module")
+def custom_cronos(tmp_path_factory):
+    path = tmp_path_factory.mktemp("cronos")
+    yield from setup_custom_cronos(
+        path, 26000, Path(__file__).parent / "configs/genesis_token_mapping.yaml"
+    )
+
+
+def test_exported_contract(custom_cronos):
+    "demonstrate that contract state can be exported to genesis"
```
Comment (yihuang) [inline integration_tests/test_exported_genesis.py:19]:
```suggestion
    "demonstrate that contract state can be deployed in genesis"
```

File: integration_tests/network.py [hunk 2]
```diff
@@ -95,3 +95,28 @@ def __init__(self, cronos, geth, contract):
...
+        "--base_port",
+        str(base_port),
+    ]
+    print(*cmd)
+    proc = subprocess.Popen(
+        cmd,
+        preexec_fn=os.setsid,
+    )
+    try:
+        wait_for_port(ports.evmrpc_port(ports.evmrpc_port(base_port)))
```
Comment (yihuang) [inline integration_tests/network.py:117]:
```suggestion
        wait_for_port(ports.evmrpc_port(base_port))
```

File: go.mod [hunk 3]
```diff
@@ -39,3 +39,6 @@ replace github.com/cosmos/ibc-go => github.com/crypto-org-chain/ibc-go v1.0.1-ho
 replace github.com/peggyjv/gravity-bridge/module => github.com/crypto-org-chain/gravity-bridge/module v0.1.22-0.20210923075944-630e7235f4b2
 
 replace github.com/cosmos/iavl => github.com/cosmos/iavl v0.17.1
+
+// FIXME: update after PR merged: https://github.com/tharsis/ethermint/pull/590
+replace github.com/tharsis/ethermint => github.com/crypto-org-chain/ethermint v0.4.2-0.20210924142240-9589b2fa9f78
```
Comment (thomas-nguy) [inline go.mod:44]: lets wait the PR to be merge so that we can use the official repo

Comment (yihuang) [reply go.mod]: done

File: CHANGELOG.md [hunk 4]
```diff
@@ -19,6 +21,7 @@ This version is the same as v0.5.3 with a patched version of ethermint which inc
 - [cronos#93](https://github.com/crypto-org-chain/cronos/pull/93) tx receipts don't contain events
 - [cronos#98](https://github.com/crypto-org-chain/cronos/pull/98) node crash under concurrent query
 
+
```
Comment (yihuang) [inline CHANGELOG.md:24]:
```suggestion
```

**Pull request merged**